src/rpc/claimtrie.cpp

```diff
@@ -180,118 +216,112 @@ bool getValueForClaim(const COutPoint& out, std::string& sValue)
 
 UniValue getvalueforname(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() != 1)
+    if (fHelp || params.size() > 2)
         throw std::runtime_error(
             "getvalueforname \"name\"\n"
             "Return the value associated with a name, if one exists\n"
             "Arguments:\n"
-            "1. \"name\"             (string) the name to look up\n"
+            "1. \"name\"          (string) the name to look up\n"
+            "2. \"blockhash\"     (string, optional) the hash of the block\n"
+            "                                           which is the active\n"
+            "                                           in the chain. If\n"
+            "                                           none is given, \n"
+            "                                           the latest block\n"
+            "                                           will be used.\n"
             "Result: \n"
```

Comment:
how about "get the value associated with the name at the block specified by this block hash. If none is given, the latest active block will be used"